x3270: 4.0ga9 -> 4.3ga8

sentientmonkey opened the pull request:
## Description of changes

* upgrade to latest release [4.3ga8 changelog](https://x3270.miraheze.org/wiki/Release_Notes/suite3270/4.3ga8)
* fixes broken darwin build by adding dependencies - fixes #304984 
* disables x11 build for darwin
* adds man page installation

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

sentientmonkey committed:
x3270: 4.0ga9 -> 4.3ga8

* upgrade to latest release
* fixes broken darwin build by adding dependancies
* disables x11 build for darwin
* adds man page installation

Co-authored-by: Sandro <sandro.jaeckel@gmail.com>

fabaff commented:
Result of `nixpkgs-review pr 304988` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>x3270</li>
  </ul>
</details>

fabaff reviewed (commented):
Could you please update the commit message from `x3270: from -> 4.0ga9 to 4.3ga8` to `x3270: 4.0ga9 -> 4.3ga8`? Thanks

anna328p commented:
curious: why the formatting changes?

sentientmonkey commented:
@fabaff will fix!

sentientmonkey commented:
@anna328p Ah, in my last PR that I had I think the suggestion was that I should use `nixpkgs-fmt` for packages, so I've been re-formatting with that when making PRs here. Happy to reformat with whichever formatter would be best.

fabaff reviewed (approved):
Change looks good to me.

fabaff commented:
@ofborg build x3270

SuperSandro2000 commented on pkgs/applications/terminal-emulators/x3270/default.nix:
```diff
@@ -1,39 +1,73 @@
+in
+stdenv.mkDerivation rec {
   pname = "x3270";
   version = "${majorVersion}.${minorVersion}${versionSuffix}";
```
Why do we construct version by hand and don't split it with lib.versions.major/minor/patch?

SuperSandro2000 commented on pkgs/applications/terminal-emulators/x3270/default.nix:
```diff
@@ -1,39 +1,73 @@
   };
 
-  buildFlags = [ "unix" ];
+  buildFlags = lib.optional stdenv.isLinux [ "unix" ];
```
```suggestion
  buildFlags = lib.optional stdenv.isLinux "unix";
```
or use optionals

SuperSandro2000 commented on pkgs/applications/terminal-emulators/x3270/default.nix:
```diff
@@ -1,39 +1,73 @@
-  buildFlags = [ "unix" ];
+  buildFlags = lib.optional stdenv.isLinux [ "unix" ];
+
+  configureFlags = lib.optional stdenv.isDarwin [
```
```suggestion
  configureFlags = lib.optionals stdenv.isDarwin [
```

SuperSandro2000 commented on pkgs/applications/terminal-emulators/x3270/default.nix:
```diff
@@ -1,39 +1,73 @@
+    make install.man
   '';
 
+  pathsToLink = [ "/share/man" ];
```
That is unused, isn't it?

sentientmonkey replied:
I added the extra `make install.man` above so that man pages are added for the builds, but I believe the pathsToLinks is needed to symlink them.

SuperSandro2000 commented on pkgs/applications/terminal-emulators/x3270/default.nix:
```diff
@@ -1,39 +1,73 @@
     ncurses
     expat
-  ];
+  ] ++ lib.optionals stdenv.isDarwin [ libiconv darwin.Security ];
```
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [ libiconv darwin.apple_sdk.frameworks.Security ];
```

sentientmonkey commented:
I squashed and rebased all of the changes so this should be ready. @SuperSandro2000 let me know if the open issues are ok if if there's anything we should do to resolve. Thanks so much for your help!

anna328p commented:
LGYM
